### PR TITLE
[mtl] expose line fill, fix MSAA views

### PR DESCRIPTION
**Pull request by kvark**
Fixes vkQuake2 running on gfx-portability.
Changes are relatively harmless.
PR checklist:
- [ ] `make` succeeds (on *nix)
- [ ] `make reftests` succeeds
- [ ] tested examples with the following backends:
- [ ] `rustfmt` run on changed code

### CODE REVIEW

**Comment by kvark** [comment]
@kondrak hi! Just a small note that MoltenVK is not the only thing capable of providing Vulkan on Apple platforms. With this fix in, https://github.com/gfx-rs/portability runs vkQuake2 fine as well.

**Comment by kondrak** [comment]
Neat! I think I might give it a shot and see what it's capable of. Great job! :+1:

**Comment by kondrak** [comment]
So looking at the patch it seems that vkQuake2 is the only title you ran into that uses lines and points? Curious!

File: src/backend/metal/src/device.rs [hunk 1]
```diff
@@ -2399,7 +2401,14 @@ impl hal::Device<Backend> for Device {
             levels: 0..raw.mipmap_level_count() as image::Level,
             layers: 0..image.kind.num_layers(),
         };
-        let mtl_type = conv::map_texture_type(kind);
+        let mtl_type = if image.mtl_type == MTLTextureType::D2Multisample {
```
Comment (grovesNL) [inline src/backend/metal/src/device.rs]: There's also `MTLTextureType2DMultisampleArray` in macOS 10.14+, I just opened #2689 to track it

File: src/backend/metal/src/conversions.rs [hunk 2]
```diff
@@ -1176,6 +1176,22 @@ pub fn map_winding(face: pso::FrontFace) -> MTLWinding {
     }
 }
 
+pub fn map_polygon_mode(mode: pso::PolygonMode) -> MTLTriangleFillMode {
+    match mode {
+        pso::PolygonMode::Point => {
+            warn!("Unable to fill with points");
+            MTLTriangleFillMode::Lines
```
Comment (grovesNL) [inline src/backend/metal/src/conversions.rs]: Is the last line removed if we submit an odd number of points?

**Comment by kvark** [comment]
Thanks @grovesNL ! Last line is not removed, since this is the fill mode, not to be confused with input geometry.

Bors r=grovesNL

**Comment by grovesNL** [comment]
Oops :) right, the amount of vertices doesn’t matter because this only applies to polygons.

**Pull request merged**